From 91df2bd6a4fcbc0fb67e9e32cb9f6a406953e011 Mon Sep 17 00:00:00 2001
Message-Id: <91df2bd6a4fcbc0fb67e9e32cb9f6a406953e011.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Shobhit Kumar <shobhit.kumar@intel.com>
Date: Mon, 15 Sep 2014 21:24:18 +0530
Subject: [PATCH 07/75] MUST_REBASE [VPG]: drm/i915/adf: Add support for i915
 pci device query from ADF

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: Ifaff75b44111f15ca499cc86f7d0dff1c4e13b90
Signed-off-by: Shobhit Kumar <shobhit.kumar@intel.com>
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |    9 +++++++++
 include/drm/i915_adf.h                  |    1 +
 2 files changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 15fc6f9..195ded7 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -59,6 +59,15 @@ void i915_adf_wrapper_teardown(void)
 	i915_adf_dev = NULL;
 }
 
+struct pci_dev *i915_adf_get_pci_dev(void)
+{
+	if (!i915_adf_dev)
+		return NULL;
+
+	return i915_adf_dev->dev->pdev;
+}
+EXPORT_SYMBOL(i915_adf_get_pci_dev);
+
 /**
  * intel_adf_pci_sideband_rw - Interface to allow ADF driver read/write to intel sideband.
  */
diff --git a/include/drm/i915_adf.h b/include/drm/i915_adf.h
index 6426ec4..02c5aec 100644
--- a/include/drm/i915_adf.h
+++ b/include/drm/i915_adf.h
@@ -40,6 +40,7 @@ extern void intel_adf_dpio_sideband_rw(u32 operation, u32 port,
 				       u32 reg, u32 *val);
 extern void intel_adf_pci_sideband_rw(u32 operation, u32 port,
 				      u32 reg, u32 *val);
+extern struct pci_dev *i915_adf_get_pci_dev(void);
 
 #endif
 
-- 
1.7.9.5

